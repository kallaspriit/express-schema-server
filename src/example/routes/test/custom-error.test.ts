import * as HttpStatus from "http-status-codes";
import * as supertest from "supertest";
import setupApp from "../../app";

let app: supertest.SuperTest<supertest.Test>;

describe("get-user", () => {
  beforeEach(async () => {
    app = supertest(await setupApp());
  });

  it("should return custom error", async () => {
    const response = await app.get("/test/custom-error").send();

    expect(response.status).toEqual(HttpStatus.BAD_REQUEST);
    expect(response.body.success).toBe(false);
    expect(response.body.payload).toBe(null);
    expect(response.body.error).toMatchSnapshot();
  });
});
