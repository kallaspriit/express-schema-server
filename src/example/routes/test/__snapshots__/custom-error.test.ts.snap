// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-user should return custom error 1`] = `"Custom error message"`;
